lib/internal/url.js

```diff
@@ -1406,12 +1410,26 @@ function getPathFromURL(path) {
 // as this is the only character that won't be percent encoded by
 // default URL percent encoding when pathname is set.
 const percentRegEx = /%/g;
-function getURLFromFilePath(filepath) {
-  const tmp = new URL('file://');
-  if (filepath.includes('%'))
-    filepath = filepath.replace(percentRegEx, '%25');
-  tmp.pathname = filepath;
-  return tmp;
+function pathToFileURL(filepath) {
+  let resolved = path.resolve(filepath);
+  // path.resolve strips trailing slashes so we must add them back
+  const filePathLast = filepath.charCodeAt(filepath.length - 1);
+  if ((filePathLast === CHAR_FORWARD_SLASH ||
+       filePathLast === CHAR_BACKWARD_SLASH) &&
+      resolved[resolved.length - 1] !== path.sep)
+    resolved += '/';
```

Comment:
☝️ Is there a case when resolve doesn't strip trailing slashes?
The check `resolved[resolved.length - 1] !== path.sep` suggests maybe?

Reply:
`path.resolve('/')` or `path.resolve('\\\\net\\x\\')`